benchmark/README.md

```diff
@@ -1,147 +1,276 @@
+[#5134](https://github.com/nodejs/node/pull/5134) pull request will be used as
+an example. This pull request _claims_ to improve the performance of the
+`string_decoder` module.
 
-For example, buffer-slice.js:
+First build two versions of node, one from the master branch (here called
+`./node-mater`) and another with the pull request applied (here called
+`./node-pr-5135`).
+
+The `compare.js` tool will then produce a csv file with the benchmark results.
 
-```bash
-node benchmark/buffers/buffer-read.js
 ```
-The output:
+$ node benchmark/compare.js string_decoder --old ./node-master --new ./node-pr-5134 > compare-pr-5134.csv
 ```
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt8 millions=1: 246.79
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt16LE millions=1: 240.11
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt16BE millions=1: 245.91
+
+For analysing the benchmark results use the `compare.R` tool.
+
+```
+$ cat compare-pr-5134.csv | Rscript benchmark/compare.R
+
+                                                                                      improvment significant      p.value
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=ascii           12.46 %         *** 1.165345e-04
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=base64-ascii    24.70 %         *** 1.820615e-15
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=base64-utf8     23.60 %         *** 2.105625e-12
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=utf8            14.04 %         *** 1.291105e-07
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=128  encoding=ascii            6.70 %           * 2.928003e-02
 ...
 ```
 
-### Run tests with options
+In the output _improvment_ is the relative improvement of the new version,
+hopefully this is positive. _significant_ tells if there are enough
+statistical evidence to validate the _improvment_. If there are enough evidence
+then there will be at least one star (`*`), more stars is just better. **However
+if there are no stars, then you shouldn't make any conclusions based on the
+_improvment_.** Sometimes this is fine, for example if you are expecting there
+to be no improvements, then there shouldn't be any stars.
+
+_For the statistically minded the R scripts makes an independent/unpaired
+2-group t-test, with the null hypothesis that the performance is the same for
+both versions. The significant field will show a star if the p-value is less
+than `0.05`._
 
-This example will run only the first type of url test, with one iteration.
-(Note: benchmarks require __many__ iterations to be statistically accurate.)
+The `compare.R` tool can also produces a box plot by using the `--plot filename`
+option. In this case there are 48 different benchmark combinations, thus you
+may want to filter the csv file. This can be during benchmarking using for
```

Comment:
s/be during/be done while/
This sentence might be totally reworded though, it sounds a bit awkward as-is. Perhaps something like:

```
This can be done while benchmarking using the `--var` parameter (e.g. `--var encoding=ascii`) or by filtering output afterwards using tools such as `sed` or `grep`.
```